### ISSUE TITLE
[BUG] "ReactSDK: Error initializing client. The core client or user promise(s) rejected."

### ISSUE THREAD

**Issue by dtrenz** [optimizely/react-sdk#235]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SDK Version

`3.0.0`

### Current Behavior

When following the instructions for implementing the React SDK for feature flags, I get this error in the JS console: 
```
ReactSDK: Error initializing client. The core client or user promise(s) rejected.
```

### Expected Behavior

No error is encountered when initializing the Optimizely React SDK.

### Steps To Reproduce

```tsx
import { createInstance, OptimizelyProvider } from "@optimizely/react-sdk";
 
const App = () => (
  <OptimizelyProvider
    optimizely={createInstance({  sdkKey: "<key>" })}
    user={{ id: "12345" }}
  >
   <Children />
  </OptimizelyProvider>
);
```

...then, in a child component:

```tsx
  const [isEnabled] = useFeature("<feature key>");
```

### React Framework

React 18.2.0 (Create React App + react-app-rewired)

### Browsers impacted

Reproducible in both Chrome and Safari. No other browsers were tested. The issue does not seem to be browser-specific.

### Logs

![image](https://github.com/optimizely/react-sdk/assets/639833/d337c79b-b6fc-422f-884f-0b4bd1c377bc)

### Severity

Minor issue

### Workaround/Solution

No, but the SDK seems to still work despite this error, so I suppose I could ignore it?

### Recent Change

This is a brand new integrations, so no.

**Comment by mikechu-optimizely** [comment]
I would complete expect No Errors even though the SDK is functional. Lemme hop on this.

**Comment by mikechu-optimizely** [comment]
Created internal ticket FSSDK-9984 and put it into next sprint.

**Comment by dominikjasek** [comment]
I am having same issue: here's a reproduction codesandbox: https://codesandbox.io/p/devbox/7ly2r6. Client is just not usable, for example I can not use `optimizelyClient.setForcedDecision()`.
![image](https://github.com/optimizely/react-sdk/assets/48070343/a5fdae7d-bc83-4b34-b32d-cd884b49eb89)


I am coming from [this issue](https://github.com/optimizely/react-sdk/issues/192), I was looking forward to the improvements, so hopefully you will manage to resolve it soon🤞

**Comment by ricky-harrys** [comment]
I'm seeing this issue as well with the same integration as @dominikjasek's codesandbox

**Comment by mikechu-optimizely** [comment]
Thanks for the sandbox @dominikjasek. I have these bugs schedule for very short priority.

**Comment by mikechu-optimizely** [comment]
I was able to reproduce this locally.

![issue-235-reproduced](https://github.com/optimizely/react-sdk/assets/104384559/96876dfc-c12d-4962-a201-9cace6e48e84)

While I investigate further, and if you're not using Advanced Audience Targeting / ODP, let's disable it explicitly. 

![issue-235-temp-fix](https://github.com/optimizely/react-sdk/assets/104384559/c6c2477b-1270-4058-a5bb-59fb0b96e99d)

I have other related GitHub issues in progress around the ODP instantiation.

**Comment by mikechu-optimizely** [comment]
Thanks for hanging in there as we align fixes for both the React and underlying JS SDKs. 

I've notified my manager that we need to look at both this and #236.  

Sorry about the low update counts here to keep you all apprised.  I will get updates and movement for you all shortly.

**Comment by mikechu-optimizely** [comment]
I'm confirming with the technical design document for Advanced Audience Targeting.  I'm questioning the need to require successful fetch across the network for the client to be considered ready. 
https://github.com/optimizely/react-sdk/blob/master/src/client.ts#L296

The explicit `odpOptions.disabled` gives us an explicit escape from AAT. 

In most cases, for now, we're more interested in audiences that are in the datafile/setup in experimentation. 

Later, we may want to block a feature or decision determination based on what we receive from the AAT/ODP/network call. ...or implementors may decide to async / useEffect additional audience qualifications into place well after the instantiation of the client to ensure performance.

WIP

cc @jaeopt

**Comment by mikechu-optimizely** [comment]
I've located a 2 part problem. 1 is in the [JS SDK](https://github.com/optimizely/javascript-sdk/pull/916).  The second is here in the React SDK.  

I'll create a new release of the JS SDK Monday then fold it into the bug fix and release for React.  

This should I fix #236.

**Comment by mikechu-optimizely** [comment]
This should be good to go.  Feel free to comment if there are updates.
